tests/infrastructure/golden_images/update_boot_source/conftest.py

```diff
@@ -125,3 +125,33 @@ def custom_data_source_scope_function(admin_client, custom_data_import_cron_scop
             f"{custom_data_import_cron_scope_function.namespace} namespace."
         )
         raise
+
+
+@pytest.fixture()
+def boot_source_data_from_data_source_dict(auto_update_data_source_matrix__function__):
+    return auto_update_data_source_matrix__function__[[*auto_update_data_source_matrix__function__][0]]
+
+
+@pytest.fixture()
+def existing_data_source_volume(
+    golden_images_persistent_volume_claims_scope_function,
+    golden_images_volume_snapshot_scope_function,
+    matrix_data_source,
+):
+    source = matrix_data_source.source
+    if source.kind == PersistentVolumeClaim.kind:
+        cluster_volumes = golden_images_persistent_volume_claims_scope_function
+    else:
+        cluster_volumes = golden_images_volume_snapshot_scope_function
+    assert any([source.name in volume.name for volume in cluster_volumes]), (
+        f"DataSource source {source.kind} {source.name} is missing"
+    )
```

Comment:
It's better to verify that the `DataSource` is `Ready: True`, and only if it's not - look at the expected source log/save its name/status

```
status:
  conditions:
  - lastHeartbeatTime: "2025-01-26T11:15:43Z"
    lastTransitionTime: "2025-01-26T11:15:43Z"
    message: DataSource is ready to be consumed
    reason: Ready
    status: "True"
    type: Ready
```